Source/LinqToDB/DataProvider/ClickHouse/ClickHouseMappingSchema.cs

```
@@ -0,0 +1,940 @@
+			// https://github.com/ClickHouse/ClickHouse/issues/38790
+			// byte[] <=> string
+			// Binary <=> string
+			SetConvertExpression((string v) => Encoding.UTF8.GetBytes(v));
+			SetConvertExpression((byte[] v) => Encoding.UTF8.GetString(v));
+			SetConvertExpression((string v) => new Binary(Encoding.UTF8.GetBytes(v)));
+			SetConvertExpression((Binary v) => Encoding.UTF8.GetString(v.ToArray()));
+			// byte[] <=> char
+			SetConvertExpression((char v) => Encoding.UTF8.GetBytes(new[] { v }));
+			SetConvertExpression((byte[] v) => Encoding.UTF8.GetChars(v).Single());
+
+			// NaN handling
+			SetConvertExpression((double v) => double.IsNaN(v) ? default : (decimal)v);
+			SetConvertExpression((float  v) => float. IsNaN(v) ? default : (decimal)v);
+			SetConvertExpression((double v) => double.IsNaN(v) ? default : (byte)v);
+			SetConvertExpression((float  v) => float. IsNaN(v) ? default : (byte)v);
+			SetConvertExpression((double v) => double.IsNaN(v) ? default : (sbyte)v);
+			SetConvertExpression((float  v) => float. IsNaN(v) ? default : (sbyte)v);
+			SetConvertExpression((double v) => double.IsNaN(v) ? default : (short)v);
+			SetConvertExpression((float  v) => float. IsNaN(v) ? default : (short)v);
+			SetConvertExpression((double v) => double.IsNaN(v) ? default : (ushort)v);
+			SetConvertExpression((float  v) => float. IsNaN(v) ? default : (ushort)v);
+			SetConvertExpression((double v) => double.IsNaN(v) ? default : (int)v);
+			SetConvertExpression((float  v) => float. IsNaN(v) ? default : (int)v);
+			SetConvertExpression((double v) => double.IsNaN(v) ? default : (uint)v);
+			SetConvertExpression((float  v) => float. IsNaN(v) ? default : (uint)v);
+			SetConvertExpression((double v) => double.IsNaN(v) ? default : (long)v);
+			SetConvertExpression((float  v) => float. IsNaN(v) ? default : (long)v);
+			SetConvertExpression((double v) => double.IsNaN(v) ? default : (ulong)v);
+			SetConvertExpression((float  v) => float. IsNaN(v) ? default : (ulong)v);
```

Is this type of silent data rewriting appropriate? I would argue that either we should throw an exception ("CH doesn't support NaN in field") or send it to CH and let them throw the exception. Silent translating of NaN to 0 to be recovered as 0 on the next query seems dangerous to me.

---

Well, actually ClickHouse supports NaN/Inf (one of 3 dbs with such support. Also supported by PostgreSQL and Firebird).

I've added this code because several standard SQL functions (some window functions) in ClickHouse return `double` but mapped in our code to decimal. This leads to exceptions from tests when we try to map them as instead of `NULL` we get `NaN` from server.

---

Ah, then I'm good here.